### DISCUSSION TITLE
Processing payment for order makes order state go to completed automatically

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3906] [Questions & Answers]
If I process bank transfers for an order I want to the order to keep the status pending/pending_payment until enough payments are processed and the order is paid in full. I then want the order to be set to processing state (for external manual processing). However I am getting a complete state automatically after $order->save();

How/where should I disable this funtionality?

**Comment by f1-outsourcing** [comment]
I saw this post[1] about this order state. If I comment this line out the order state is not automatically set to completed. But I actually only have a few products that need some external activities, before they can be set to completed.


```

   2390     protected function _checkState()
   2391     {
   2392         if (!$this->getId()) {
   2393             return $this;
   2394         }
   2395
   2396         $userNotification = $this->hasCustomerNoteNotify() ? $this->getCustomerNoteNotify() : null;
   2397
   2398         if (!$this->isCanceled()
   2399             && !$this->canUnhold()
   2400             && !$this->canInvoice()
   2401             && !$this->canShip()
   2402         ) {
   2403             if ($this->getBaseGrandTotal() == 0 || $this->canCreditmemo()) {
   2404                 if ($this->getState() !== self::STATE_COMPLETE) {

   2406                     //$this->_setState(self::STATE_COMPLETE, true, '', $userNotification);
   2407                     $this->_setState(self::STATE_PROCESSING, true, '', $userNotification);
   2408                 }
   2409             } elseif ((float) $this->getTotalRefunded() || (!$this->getTotalRefunded()
```

[1]
https://stackoverflow.com/questions/28645095/magento-disable-auto-state-complete-for-virtual-orders

**Comment by f1-outsourcing** [comment]
I don't really understand how this automatic setting should work. If the order total is 10, and I process a payment for 10, the order goes automatically to COMPLETE state. However if I have an order with total 10 and I process to separate payments of 5, no COMPLETE state is applied.

**Comment by kiatng** [comment]
I cannot replicate it, I placed an order from frontend with a total of $2.00, the order status is Pending:
![image](https://github.com/OpenMage/magento-lts/assets/1106470/bed71d1c-0617-4246-9f4e-9a845e59c495)
